lib/skate_web/controllers/auth_controller.ex

```diff
@@ -28,10 +30,10 @@ defmodule SkateWeb.AuthController do
 
   def callback(%{assigns: %{ueberauth_failure: _fails}} = conn, _params) do
     # Users are sometimes seeing unexpected Ueberauth failures of unknown provenance.
-    # Instead of sending a 403 unauthenticated response, we are signing them out and
-    # sending them to the home page to start the auth path over again.
-    # We should be on the lookout for users getting trapped in a loop because of this.
-    # If we observe that happening we should rethink this remedy. -- MSS 2019-07-03
+    # Instead of sending a 403 unauthenticated response immediately, we are signing them out and
+    # sending them to the home page to start the auth path over again. -- MSS 2019-07-03
+    # We are maintaining the retry logic but limiting it to only a set number of retries using
+    # logic in SkateWeb.AuthManager.ErrorHandler. -- LEM 2023-04-27
```

Comment:
question: Should this comment & the retry logic be on the same function? I'm a bit confused as to how this `callback` function relates to the `error_handler`

Reply:
Okay, so this is part of what's so confusing about this logic. 😅 

`AuthController` handles the HTTP request that is made in the callback phase of the auth flow handled by Ueberauth - this is the point where Cognito redirects the user's browser back to Skate with some information about the login flow, ideally a signed assertion that the logic was successful with further metadata. The reason that the comment is here is that usually a failure would mean that we would return an unauthorized response, and the part that's weird is that we're redirecting back to the homepage instead.

`ErrorHandler` is part of our usage of Guardian. This is the functionality that is invoked when a user attempts to access a page requiring authentication (i.e. most of Skate) but there is no current valid Guardian ticket. Since we use Ueberauth to handle the authentication step the usual way to handle this state is to redirect to the request phase of the auth flow. My initial impression had been that the retry logic would actually be easier to implement cleanly in `ErrorHandler` since that's the start of the whole auth flow rather than an intermediate step but now that I think about it it would probably be doable as well. I guess it's a tradeoff in terms of what we think is most readable.